assets/lokomotive-kubernetes/bootkube/resources/bootstrap-manifests/bootstrap-controller-manager.yaml

```diff
@@ -18,6 +18,8 @@ spec:
     - --cluster-signing-cert-file=/etc/kubernetes/secrets/ca.crt
     - --cluster-signing-key-file=/etc/kubernetes/secrets/ca.key
     - --configure-cloud-routes=false
+    - --controllers=*,tokencleaner,bootstrapsigner
+    - --experimental-cluster-signing-duration=1h
```

Comment:
What should be the ideal time to renew the certificate? Days/Hours? If hours how many?

Reply:
Why not use default?

Reply:
Also, the `--controllers` flag is not really needed I think. Can you elaborate why you add it?

Reply:
Docs mention that you should add it. So here https://kubernetes.io/docs/reference/access-authn-authz/bootstrap-tokens/#enabling-bootstrap-token-authentication.

Also 

> --controllers stringSlice     Default: [*]
> 
>  | A list of controllers to enable. '*' enables all on-by-default controllers, 'foo' enables the controller named 'foo', '-foo' disables the controller named 'foo'.All controllers: attachdetach, bootstrapsigner, cloud-node-lifecycle, clusterrole-aggregation, cronjob, csrapproving, csrcleaner, csrsigning, daemonset, deployment, disruption, endpoint, endpointslice, garbagecollector, horizontalpodautoscaling, job, namespace, nodeipam, nodelifecycle, persistentvolume-binder, persistentvolume-expander, podgc, pv-protection, pvc-protection, replicaset, replicationcontroller, resourcequota, root-ca-cert-publisher, route, service, serviceaccount, serviceaccount-token, statefulset, tokencleaner, ttl, ttl-after-finished Disabled-by-default controllers: bootstrapsigner, tokencleaner

src: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/

see the last line says that `Disabled-by-default controllers: bootstrapsigner, tokencleaner` are disabled by default and we need to enable them.

Reply:
Right. I'm not exactly sure how this configmap signing process works, but it seems it's something we should use :+1: thanks for clarifying.

Reply:
It turns out we don't need `bootstrapsigner`, it is a kubeadm feature. Since in kubeadm the way for a worker machine to join the cluster is by using the token, there is no way for the worker to know if the apiserver it is trying to join is a valid one and thus this feature comes in handy.

Also the worker node has no CA data and this configmap helps in acquiring that as well. We don't have such problems though.